resources/lib/kodi/utils.py

```diff
@@ -45,8 +45,10 @@ def set_setting(setting, value):
 
 
 def get_device_id():
-    return get_infolabel("System.FriendlyName")
-
+    friendly_name = get_infolabel("System.FriendlyName")
+    if not friendly_name:
+        friendly_name = "Kodi TV"
```

Comment:
maybe add it to a string setting so that the user can configure it? Define a enum:
- From kodi
- Custom

Custom is used if forced on settings or if the first infobool fails

Reply:
Unfortunately I am not very familiar with both python an kodi settings, so it's the easiest fix of existing problem, when discovery is not working in youtube app, when friendly name is null.

Any name - is better than no name at all (in cases when FriendlyName is not provided by environment)